<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
</head>

<body>
  <p>123</p>
  <script>
    function fn() {
      try {
        debugger
        // 可能发生错误的代码，要写到 try 里面
        const p = document.querySelector('p')
        p.style.color = 'red'
      } catch (err) { // err保存了浏览器提供的信息
        // 拦截错误，提示浏览器提供的错误信息，但是不中断程序的执行（如果没有错误，不执行catch(){}里的代码）
        console.log(err.message) // message属性
        // 抛出异常，中断程序
        throw new Error('选择器错误')
        // 需要加return，中断程序，这里可以结合throw抛异常
        // return 
      } finally {
        // 不管程序有没有错误，最终都会执行finally里的代码
        alert('弹出对话框')
      }

      console.log('没被中断')
    }

    fn() // 调用
  </script>
</body>

</html>